### DISCUSSION TITLE
Embedding molecules with unchemical aromaticity

### DISCUSSION THREAD

**Discussion by NiklasKappel** [rdkit/rdkit#5394] [Q&A] [answered]
I am trying to use `RDKit::DGeomHelpers::EmbedMolecule` to generate a 3d conformer for the "molecule" given by the SMILES `cN(c)c`. I realize that this is not a chemically correct molecule, because each carbon atom has three unmatched bonds and because the carbon atoms are marked as aromatic even though they do not appear to be in a ring structure. However, `cN(c)c` is a meaningful SMILES in the sense that it represents a substructure of a real molecule, e.g. [TPD](https://www.chemspider.com/Chemical-Structure.93294.html). This notion is shared by `RDKit::SubstructMatch`.

Unfortunately, calling `EmbedMolecule` on a molecule loaded from that SMILE throws an `RDKit::AtomKekulizeException`: "non-ring atom 0 marked aromatic". Here is my code in detail:

```cpp
#include <memory>
#include <string>

#include <rdkit/GraphMol/DistGeomHelpers/Embedder.h>
#include <rdkit/GraphMol/GraphMol.h>
#include <rdkit/GraphMol/MolOps.h>
#include <rdkit/GraphMol/SmilesParse/SmilesParse.h>

auto embed_molecule(RDKit::ROMol& molecule) -> void {
  auto params =
      RDKit::DGeomHelpers::EmbedParameters{RDKit::DGeomHelpers::ETKDG};
  params.randomSeed = 1234;
  RDKit::DGeomHelpers::EmbedMolecule(molecule, params);  // this line throws!
}

auto smile_to_radical(std::string const& smile)
    -> std::unique_ptr<RDKit::RWMol> {
  auto params = RDKit::SmilesParserParams{};
  params.removeHs = false;
  params.sanitize = false;
  auto radical =
      std::unique_ptr<RDKit::RWMol>{RDKit::SmilesToMol(smile, params)};
  radical->updatePropertyCache();
  radical->getRingInfo()->initialize();
  return radical;
}

auto smile_to_3d_radical(std::string const& smile)
    -> std::unique_ptr<RDKit::RWMol> {
  auto radical = smile_to_radical(smile);
  embed_molecule(*radical);
  return radical;
}

int main() {
  auto mymol = smile_to_3d_radical("cN(c)c");
}
```

Note that `smile_to_radical("cN(c)c")` (without embedding) yields a well behaved `RDKit::RWMol` that can be successfully substructure-matched to a regular TPD molecule. Is there a possibility to ignore the exception and still generate a conformer, e.g. a setting to turn off any kekulization or sanitation attempts?

Edit: I'm using RDKit version 2022.03.3 installed with conda from the conda-forge channel.

**Comment by greglandrum** [comment, marked as answer]
It seems like this should work, but I will need to take a look at the code when I am back in the office next week

**Reply by greglandrum** [reply]
@NiklasKappel : By default the embedding code splits the molecule into fragments and then embeds the fragments separately. The individual fragments are sanitized after splitting. The easiest fix for you is to just skip that splitting step by setting: `params.embedFragmentsSeparately = false`

An aside: in order to have a chance at getting decent conformers, you really should add the appropriate Hs to the molecule and set the hybridization states of the atoms correctly.
